File: mm/dailyTemperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {
  const answer = new Array(temperatures.length).fill(0);
  const stack = [];

  for (let i = 0; i < temperatures.length; i++) {
    while (stack.length && temperatures[i] > stack[stack.length - 1][1]) {
      const [index] = stack.pop();
      answer[index] = i - index;
    }

    stack.push([i, temperatures[i]]);
  }

  return answer;
};

File: JavaWorkSpace/workspace/src/JavaPlayGround/DailyTemperatures.java
package JavaPlayGround;

import java.util.Stack;

public class DailyTemperatures {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] output = new int[temperatures.length];
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int index = stack.pop();
                output[index] = i - index;
            }
            stack.push(i);
        }

        return output;
    }
}
